cognitive/src/main/scala/com/microsoft/azure/synapse/ml/cognitive/TextToSpeechSSMLGenerator.scala

```diff
@@ -0,0 +1,113 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.azure.synapse.ml.cognitive
+
+import com.microsoft.azure.synapse.ml.cognitive.TextToSpeechSSMLFormat.TextToSpeechSSMLResponseFormat
+import com.microsoft.azure.synapse.ml.logging.BasicLogging
+import org.apache.http.client.methods.HttpRequestBase
+import org.apache.http.entity.{AbstractHttpEntity, StringEntity}
+import org.apache.spark.ml.{ComplexParamsReadable, NamespaceInjections, PipelineModel, Transformer}
+import org.apache.spark.ml.param.ParamMap
+import org.apache.spark.ml.util.Identifiable
+import org.apache.spark.sql.catalyst.encoders.RowEncoder
+import org.apache.spark.sql.types.{StringType, StructType}
+import org.apache.spark.sql.{DataFrame, Dataset, Row}
+import spray.json._
+
+object TextToSpeechSSMLGenerator extends ComplexParamsReadable[TextToSpeechSSMLGenerator] with Serializable
+class TextToSpeechSSMLGenerator(override val uid: String) extends CognitiveServicesBase(uid)
+  with HasSSMLGeneratorParams with HasCognitiveServiceInput
+  with HasInternalStringOutputParser
+  with BasicLogging {
+  logClass()
+
+  def this() = this(Identifiable.randomUID(classOf[TextToSpeechSSMLGenerator].getSimpleName))
+
+  setDefault(errorCol -> (uid + "_errors"))
+  setDefault(locale -> Left("en-US"))
+  setDefault(voice -> Left("en-US-SaraNeural"))
+
+  def urlPath: String = "cognitiveservices/v1"
+
+  protected val additionalHeaders: Map[String, String] = Map[String, String](
+    ("X-Microsoft-OutputFormat", "textanalytics-json"),
+    ("Content-Type", "application/ssml+xml")
+  )
+
+  override def inputFunc(schema: StructType): Row => Option[HttpRequestBase] = super.inputFunc(schema).andThen(r => {
+    r.map(req => {
+      additionalHeaders.foreach(header => req.setHeader(header._1, header._2))
+      req
+    })
+  })
+
+  override protected def prepareEntity: Row => Option[AbstractHttpEntity] = { row =>
+    val localeValue = getValue(row, locale)
+    val zhCNVoiceName = "Microsoft Server Speech Text to Speech Voice (zh-CN, XiaomoNeural)"
+    val enUSVoiceName = "Microsoft Server Speech Text to Speech Voice (en-US, JennyNeural)"
+    val voiceName = if (localeValue == "zh-CN") zhCNVoiceName else enUSVoiceName
+    val textValue = getValue(row, text)
+    val body: String =
+      s"<speak version='1.0' xmlns='http://www.w3.org/2001/10/synthesis'" +
+        s" xmlns:mstts='https://www.w3.org/2001/mstts'" +
+        s" xml:lang='${localeValue}'><voice xml:lang='${localeValue}' xml:gender='Female' name='${voiceName}'>" +
```

Comment:
likewise gender='Female' here is a little specific, is this necessary or would it infer this automatically given the voice name?

Reply:
Addressed in most recent commit. Only supporting single speaker for now and things like gender="Female" are not necessary